siliconcompiler/scheduler.py

```diff
@@ -16,63 +17,87 @@ def _deferstep(chip, step, index, status):
 
     # Determine which HPC job scheduler being used.
     scheduler_type = chip.get('option', 'jobscheduler')
-    username = chip.status['slurm_account']
-    partition = chip.status['slurm_partition']
+
+    # Determine which cluster parititon to use. (Default value can be overridden on per-step basis)
+    if f'{step}_slurm_partition' in chip.status:
+        partition = chip.status[f'{step}_slurm_partition']
+    else:
+        partition = chip.status['slurm_partition']
+    # Get the temporary UID associated with this job run.
     job_hash = chip.status['jobhash']
+
+    # Job data is not encrypted, so it can be run in shared storage.
+    # Write out the current schema for the compute node to pick up.
+    job_dir = chip._getworkdir()
+    cfg_dir = f'{job_dir}/configs'
+    cfg_file = f'{cfg_dir}/{step}{index}.json'
+    if not os.path.isdir(cfg_dir):
+        os.mkdir(cfg_dir)
+    chip.write_manifest(cfg_file)
+
     if scheduler_type == 'slurm':
         # The script defining this Chip object may specify feature(s) to
         # ensure that the job runs on a specific subset of available nodes.
         if 'slurm_constraint' in chip.status:
             slurm_constraint = chip.status['slurm_constraint']
         else:
             slurm_constraint = 'SHARED'
+
         # Set the log file location.
+        # TODO: May need to prepend ('option', 'builddir') and remove the '--chdir' arg if
+        # running on a locally-managed cluster control node instead of submitting to a server app.
+        #output_file = os.path.join(chip.get('option', 'builddir'),
         output_file = os.path.join(chip.get('design'),
                                    chip.get('option', 'jobname'),
                                    f'sc_remote-{step}-{index}.log')
         schedule_cmd = ['sbatch', '--exclusive',
                        '--constraint', slurm_constraint,
-                       '--account', username,
                        '--partition', partition,
                        '--chdir', chip.get('option', 'builddir'),
                        '--job-name', f'{job_hash}_{step}{index}',
-                       '--output', output_file]
+                       '--output', output_file,
+                       ]
+        # Only specify an account if accounting is required for this cluster/run.
+        if 'slurm_account' in chip.status:
+            username = chip.status['slurm_account']
+            schedule_cmd.extend(['--acount', username])
     elif scheduler_type == 'lsf':
         # TODO: LSF support is untested and currently unsupported.
         schedule_cmd = ['lsrun']
 
-    # Job data is not encrypted, so it can be run in shared storage.
-    # Write out the current schema for the compute node to pick up.
-    job_dir = chip._getworkdir()
-    cfg_dir = f'{job_dir}/configs'
-    cfg_file = f'{cfg_dir}/{step}{index}.json'
-    if not os.path.isdir(cfg_dir):
-        os.mkdir(cfg_dir)
-    chip.write_manifest(cfg_file)
+    time.sleep(1.0)
```

Comment:
Whe the need for sleep here?

Reply:
There shouldn't be a need for it anymore; removed.

I think that I originally inserted this delay to give NFS caches some time to catch up after seeing some unwarranted 'file not found' errors. The long-term fix for that was to add prolog/epilog scripts which refresh the mounts before and after each job, though.